### PR TITLE
feat: ENT-5042 | added validation on sales_force_id in Enterprise Offers.

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/enterprise/forms.py [hunk 1]
```diff
@@ -149,6 +151,16 @@ def clean_max_global_applications(self):
 
         return self.cleaned_data.get('max_global_applications')
 
+    def clean_sales_force_id(self):
+        # validate sales_force_id format
+        sales_force_id = self.cleaned_data.get('sales_force_id')
+        if not re.match(ENTERPRISE_SALES_FORCE_ID_REGEX, sales_force_id):
+            self.add_error(
+                'sales_force_id',
+                _('Salesforce Opportunity ID must be 18 alphanumeric characters and begin with 006.')
```
Comment (sameenfatima78) [inline ecommerce/enterprise/forms.py:160]: Do all opp IDs start with 006?

**Review by sameenfatima78 — APPROVED** [review]
Just one question.. otherwise looks good

File: ecommerce/enterprise/constants.py [hunk 2]
```diff
@@ -13,3 +13,5 @@
 # Default Sender Alias used in Enterprise Customer Code Assign,Remind and Revoke Emails.
 
 SENDER_ALIAS = 'edX Support Team'
+
+ENTERPRISE_SALES_FORCE_ID_REGEX = r'^006[a-zA-Z0-9]{15}$|^none$'
```
Comment (muhammad-ammar) [inline ecommerce/enterprise/constants.py:19]: can you please add a comment to explain what kind of strings this regex will work for?

**Review by muhammad-ammar — APPROVED** [review]
LGTM. Just one comment.

File: ecommerce/enterprise/constants.py (hunk 2 above)
Comment (HammadAhmadWaqas) [reply ecommerce/enterprise/constants.py:19]: @muhammad-ammar sure

File: ecommerce/enterprise/forms.py (hunk 1 above)
Comment (HammadAhmadWaqas) [reply ecommerce/enterprise/forms.py:160]: @sameenfatima78 yes this what requirement says

**Pull request merged**